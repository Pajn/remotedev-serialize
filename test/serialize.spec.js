var Immutable = require('immutable');
var Serialize = require('../');
var serialize = Serialize(Immutable);
var stringify = serialize.stringify;
var parse = serialize.parse;

var data = {
  map: Immutable.Map({ a: 1, b: 2, c: 3, d: 4 }),
  orderedMap: Immutable.OrderedMap({ a: 1, b: 2, c: 3, d: 4 }),
  list: Immutable.List([1,2,3,4,5,6,7,8,9,10]),
  range: Immutable.Range(0,7),
  repeat: Immutable.Repeat('hi', 100),
  set: Immutable.Set([10,9,8,7,6,5,4,3,2,1]),
  orderedSet: Immutable.OrderedSet([10,9,8,7,6,5,4,3,2,1]),
  seq: Immutable.Seq.of(1,2,3,4,5,6,7,8),
  stack: Immutable.Stack.of('a', 'b', 'c')
};

var stringified = {};
describe('Stringify', function () {
  Object.keys(data).forEach(function (key) {
    it(key, function () {
      stringified[key] = stringify(data[key]);
      expect(stringified[key]).toMatchSnapshot();
    });
  });
});

describe('Parse', function () {
  Object.keys(data).forEach(function(key) {
    it(key, function() {
      expect(parse(stringified[key])).toEqual(data[key]);
    });
  });
});

describe('Record', function () {
  var ABRecord = Immutable.Record({ a:1, b:2 });
  var myRecord = new ABRecord({ b:3 });
  
  var serialize = Serialize(Immutable, [ABRecord]);
  var stringify = serialize.stringify;
  var parse = serialize.parse;
  var stringifiedRecord;

  it('stringify', function() {
    stringifiedRecord = stringify(myRecord);
    expect(stringifiedRecord).toMatchSnapshot();
  });

  it('parse', function() {
    stringifiedRecord = stringify(myRecord);
    expect(parse(stringifiedRecord)).toEqual(myRecord);
  });
});
