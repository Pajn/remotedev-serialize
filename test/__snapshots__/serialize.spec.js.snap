exports[`Record stringify 1`] = `"{\"data\":{\"a\":1,\"b\":3},\"__remotedevType__\":\"ImmutableRecord\",\"__remotedevRef__\":0}"`;

exports[`Stringify list 1`] = `"{\"data\":[1,2,3,4,5,6,7,8,9,10],\"__remotedevType__\":\"ImmutableList\"}"`;

exports[`Stringify map 1`] = `"{\"data\":{\"a\":1,\"b\":2,\"c\":3,\"d\":4},\"__remotedevType__\":\"ImmutableMap\"}"`;

exports[`Stringify orderedMap 1`] = `"{\"data\":{\"a\":1,\"b\":2,\"c\":3,\"d\":4},\"__remotedevType__\":\"ImmutableOrderedMap\"}"`;

exports[`Stringify orderedSet 1`] = `"{\"data\":[10,9,8,7,6,5,4,3,2,1],\"__remotedevType__\":\"ImmutableOrderedSet\"}"`;

exports[`Stringify range 1`] = `"{\"data\":{\"_start\":0,\"_end\":7,\"_step\":1,\"size\":7},\"__remotedevType__\":\"ImmutableRange\"}"`;

exports[`Stringify repeat 1`] = `"{\"data\":{\"_value\":\"hi\",\"size\":100},\"__remotedevType__\":\"ImmutableRepeat\"}"`;

exports[`Stringify seq 1`] = `"{\"data\":[1,2,3,4,5,6,7,8],\"__remotedevType__\":\"ImmutableSeq\"}"`;

exports[`Stringify set 1`] = `"{\"data\":[1,2,3,4,5,6,7,8,9,10],\"__remotedevType__\":\"ImmutableSet\"}"`;

exports[`Stringify stack 1`] = `"{\"data\":[\"a\",\"b\",\"c\"],\"__remotedevType__\":\"ImmutableStack\"}"`;
